AppWidget/app/src/main/java/com/example/android/appwidget/ItemsCollectionAppWidget.kt

```diff
@@ -0,0 +1,185 @@
+        ) {
+            intent.extras?.let {
+                // The checked state is stored in the Bundle in the Intent
+                val checked = it.getBoolean(
+                    RemoteViews.EXTRA_CHECKED,
+                    false
+                )
+                Toast.makeText(
+                    context,
+                    "ViewId : ${it.getInt(EXTRA_VIEW_ID)}'s checked status is now : $checked",
+                    Toast.LENGTH_SHORT
+                ).show()
+            }
+        }
+    }
+}
+
+class ItemsCollectionRemoteViewsService : RemoteViewsService() {
+
+    override fun onGetViewFactory(data: Intent?): RemoteViewsFactory {
+        return ItemsCollectionRemoteViewsFactory(applicationContext)
+    }
+}
+
+class ItemsCollectionRemoteViewsFactory(private val context: Context) :
+    RemoteViewsService.RemoteViewsFactory {
+
+    companion object {
+        val items = listOf(
+            R.layout.item_checkboxes,
+            R.layout.item_radio_buttons,
+            R.layout.item_switches,
+        )
+
+        const val REQUEST_CODE_FROM_COLLECTION_WIDGET = 2
+        const val EXTRA_VIEW_ID = "extra_view_id"
+        const val REQUEST_CODE = "request_code"
+
+        @RequiresApi(31)
+        fun getRemoteCollectionItems(context: Context): RemoteViews.RemoteCollectionItems {
+            val builder = RemoteViews.RemoteCollectionItems.Builder()
+            items.forEachIndexed { index, layoutId ->
+                builder.addItem(index.toLong(), getRemoteViewsForLayoutId(context, layoutId))
+            }
+            return builder.setHasStableIds(true).setViewTypeCount(items.count()).build()
+        }
+
+        internal fun getRemoteViewsForLayoutId(
+            context: Context,
+            @LayoutRes layoutId: Int
+        ): RemoteViews {
+            val remoteViews = RemoteViews(context.packageName, layoutId)
+            if (BuildCompat.isAtLeastS()) {
+                // Compound buttons in a widget are stateless. You need to change the state and register for
+                // the state change events.
+                when (layoutId) {
+                    R.layout.item_checkboxes -> {
+                        // This code will check the Checkbox
+                        remoteViews.setCompoundButtonChecked(R.id.item_checkbox, true)
+                    }
+                    R.layout.item_radio_buttons -> {
+                        // This code will check the item_radio_button2 in the item_radio_group RadioGroup
+                        remoteViews.setRadioGroupChecked(
+                            R.id.item_radio_group,
+                            R.id.item_radio_button2
+                        )
+                    }
+                    R.layout.item_switches -> {
+                        val viewId = R.id.item_switch
+                        val onCheckedChangePendingIntent = PendingIntent.getBroadcast(
```

Comment:
I think we should register a on checked change listener for the checkboxes and radio buttons as well. The `RemoteViews.setRadioGroupChecked()` and `RemoteViews.setCompoundButtonChecked()` APIs are fairly straightforward, so I think it would be more beneficial to demonstrate how to listen for state changes.

Reply:
Let me do that in a follow up pull request since landing this pull request is time bound. And we are demonstrating how to listen for the state change.